src/sentry/api/endpoints/project_transaction_threshold.py

```diff
@@ -90,29 +89,15 @@ def post(self, request: Request, project) -> Response:
 
         data = serializer.validated_data
 
-        with transaction.atomic(router.db_for_write(ProjectTransactionThreshold)):
-            try:
-                project_threshold = ProjectTransactionThreshold.objects.get(
-                    project=project,
-                    organization=project.organization,
-                )
-                project_threshold.threshold = data.get("threshold") or project_threshold.threshold
-                project_threshold.metric = data.get("metric") or project_threshold.metric
-                project_threshold.edited_by_id = request.user.id
-                project_threshold.save()
-
-                created = False
-
-            except ProjectTransactionThreshold.DoesNotExist:
-                project_threshold = ProjectTransactionThreshold.objects.create(
-                    project=project,
-                    organization=project.organization,
-                    threshold=data.get("threshold", 300),
-                    metric=data.get("metric", TransactionMetric.DURATION.value),
-                    edited_by_id=request.user.id,
-                )
-
-                created = True
+        project_threshold, created = ProjectTransactionThreshold.objects.update_or_create(
+            project=project,
+            organization=project.organization,
+            defaults={
+                "threshold": data["threshold"],
```

Comment:
are we no longer setting a threshold by default?

Reply:
threshold is a required field (it's validated by the [serializer](https://github.com/getsentry/sentry/pull/68945/files#diff-b3719818ff76344df63e472fee6715c13603c80806f4150ccbfdeb3f7de9408aR90)) defaults are hardcoded if the project doesn't have one set

Reply:
Looking at the serializer, it's not a required field and it does not specify a default either https://github.com/getsentry/sentry/blob/95862bb96d08541a3c1ec36efdc6cf55b76ac8e8/src/sentry/api/endpoints/project_transaction_threshold.py#L26

Reply:
gosh darnit, I was looking at the override one https://github.com/getsentry/sentry/blob/master/src/sentry/api/endpoints/project_transaction_threshold_override.py#L23